commit a594533df0f6ca391da003f43d53b336a2d23ffa
Merge: cdb9d3537711 66efff515a65
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Tue Dec 13 11:59:58 2022 -0800

    Merge tag 'drm-next-2022-12-13' of git://anongit.freedesktop.org/drm/drm
    
    Pull drm updates from Dave Airlie:
     "The biggest highlight is that the accel subsystem framework is merged.
      Hopefully for 6.3 we will be able to line up a driver to use it.
    
      In drivers land, i915 enables DG2 support by default now, and nouveau
      has a big stability refactoring and initial ampere support, AMD
      includes new hw IP support and should build on ARM again. There is
      also an ofdrm driver to take over offb on platforms it's used.
    
      Stuff outside my tree, the dma-buf patches hit a few places, the vc4
      firmware changes also do, and i915 has some interactions with MEI for
      discrete GPUs. I think all of those should have been acked/reviewed by
      relevant parties.
    
      New driver:
       - ofdrm - replacement for offb
    
      fbdev:
       - add support for nomodeset
    
      fourcc:
       - add Vivante tiled modifier
    
      core:
       - atomic-helpers: CRTC primary plane test fixes, fb access hooks
       - connector: TV API consistency, cmdline parser improvements
       - send connector hotplug on cleanup
       - sort makefile objects
    
      tests:
       - sort kunit tests
       - improve DP-MST tests
       - add kunit helpers to create a device
    
      sched:
       - module param for scheduling policy
       - refcounting fix
    
      buddy:
       - add back random seed log
    
      ttm:
       - convert ttm_resource to size_t
       - optimize pool allocations
    
      edid:
       - HFVSDB parsing support fixes
       - logging/debug improvements
       - DSC quirks
    
      dma-buf:
       - Add unlocked vmap and attachment mapping
       - move drivers to common locking convention
       - locking improvements
    
      firmware:
       - new API for rPI firmware and vc4
    
      xilinx:
       - zynqmp: displayport bridge support
       - dpsub fix
    
      bridge:
       - adv7533: Remove dynamic lane switching
       - it6505: Runtime PM support, sync improvements
       - ps8640: Handle AUX defer messages
       - tc358775: Drop soft-reset over I2C
    
      panel:
       - panel-edp: Add INX N116BGE-EA2 C2 and C4 support.
       - Jadard JD9365DA-H3
       - NewVision NV3051D
    
      amdgpu:
       - DCN support on ARM
       - DCN 2.1 secure display
       - Sienna Cichlid mode2 reset fixes
       - new GC 11.x firmware versions
       - drop AMD specific DSC workarounds in favour of drm code
       - clang warning fixes
       - scheduler rework
       - SR-IOV fixes
       - GPUVM locking fixes
       - fix memory leak in CS IOCTL error path
       - flexible array updates
       - enable new GC/PSP/SMU/NBIO IP
       - GFX preemption support for gfx9
    
      amdkfd:
       - cache size fixes
       - userptr fixes
       - enable cooperative launch on gfx 10.3
       - enable GC 11.0.4 KFD support
    
      radeon:
       - replace kmap with kmap_local_page
       - ACPI ref count fix
       - HDA audio notifier support
    
      i915:
       - DG2 enabled by default
       - MTL enablement work
       - hotplug refactoring
       - VBT improvements
       - Display and watermark refactoring
       - ADL-P workaround
       - temp disable runtime_pm for discrete-
       - fix for A380 as a secondary GPU
       - Wa_18017747507 for DG2
       - CS timestamp support fixes for gen5 and earlier
       - never purge busy TTM objects
       - use i915_sg_dma_sizes for all backends
       - demote GuC kernel contexts to normal priority
       - gvt: refactor for new MDEV interface
       - enable DC power states on eDP ports
       - fix gen 2/3 workarounds
    
      nouveau:
       - fix page fault handling
       - Ampere acceleration support
       - driver stability improvements
       - nva3 backlight support
    
      msm:
       - MSM_INFO_GET_FLAGS support
       - DPU: XR30 and P010 image formats
       - Qualcomm SM6115 support
       - DSI PHY support for QCM2290
       - HDMI: refactored dev init path
       - remove exclusive-fence hack
       - fix speed-bin detection
       - enable clamp to idle on 7c3
       - improved hangcheck detection
    
      vmwgfx:
       - fb and cursor refactoring
       - convert to generic hashtable
       - cursor improvements
    
      etnaviv:
       - hw workarounds
       - softpin MMU fixes
    
      ast:
       - atomic gamma LUT support
       - convert to SHMEM
    
      lcdif:
       - support YUV planes
       - Increase DMA burst size
       - FIFO threshold tuning
    
      meson:
       - fix return type of cvbs mode_valid
    
      mgag200:
       - fix PLL setup on some revisions
    
      sun4i:
       - A100 and D1 support
    
      udl:
       - modesetting improvements
       - hot unplug support
    
      vc4:
       - support PAL-M
       - fix regression preventing 4K @ 60Hz
       - fix NULL ptr deref
    
      v3d:
       - switch to drm managed resources
    
      renesas:
       - RZ/G2L DSI support
       - DU Kconfig cleanup
    
      mediatek:
       - fixup dpi and hdmi
       - MT8188 dpi support
       - MT8195 AFBC support
    
      tegra:
       - NVDEC hardware on Tegra234 SoC
    
      hdlcd:
       - switch to drm managed resources
    
      ingenic:
       - fix registration error path
    
      hisilicon:
       - convert to drm_mode_init
    
      maildp:
       - use managed resources
    
      mtk:
       - use drm_mode_init
    
      rockchip:
       - use drm_mode_copy"
    
    * tag 'drm-next-2022-12-13' of git://anongit.freedesktop.org/drm/drm: (1397 commits)
      drm/amdgpu: fix mmhub register base coding error
      drm/amdgpu: add tmz support for GC IP v11.0.4
      drm/amdgpu: enable GFX Clock Gating control for GC IP v11.0.4
      drm/amdgpu: enable GFX Power Gating for GC IP v11.0.4
      drm/amdgpu: enable GFX IP v11.0.4 CG support
      drm/amdgpu: Make amdgpu_ring_mux functions as static
      drm/amdgpu: generally allow over-commit during BO allocation
      drm/amd/display: fix array index out of bound error in DCN32 DML
      drm/amd/display: 3.2.215
      drm/amd/display: set optimized required for comp buf changes
      drm/amd/display: Add debug option to skip PSR CRTC disable
      drm/amd/display: correct DML calc error of UrgentLatency
      drm/amd/display: correct static_screen_event_mask
      drm/amd/display: Ensure commit_streams returns the DC return code
      drm/amd/display: read invalid ddc pin status cause engine busy
      drm/amd/display: Bypass DET swath fill check for max clocks
      drm/amd/display: Disable uclk pstate for subvp pipes
      drm/amd/display: Fix DCN2.1 default DSC clocks
      drm/amd/display: Enable dp_hdmi21_pcon support
      drm/amd/display: prevent seamless boot on displays that don't have the preferred dig
      ...

diff --cc drivers/gpu/drm/vmwgfx/vmwgfx_scrn.c
index 9c79873f62f0,8db61c541a80..e1f36a09c59c
--- a/drivers/gpu/drm/vmwgfx/vmwgfx_scrn.c
+++ b/drivers/gpu/drm/vmwgfx/vmwgfx_scrn.c
@@@ -947,12 -924,8 +924,12 @@@ err_free
  int vmw_kms_sou_init_display(struct vmw_private *dev_priv)
  {
  	struct drm_device *dev = &dev_priv->drm;
- 	int i, ret;
+ 	int i;
  
 +	/* Screen objects won't work if GMR's aren't available */
 +	if (!dev_priv->has_gmr)
 +		return -ENOSYS;
 +
  	if (!(dev_priv->capabilities & SVGA_CAP_SCREEN_OBJECT_2)) {
  		return -ENOSYS;
  	}